presto-spi/src/main/java/com/facebook/presto/spi/block/LongArrayBlockBuilder.java

```diff
@@ -87,6 +88,7 @@ public BlockBuilder appendNull()
 
         valueIsNull[positionCount] = true;
 
+        hasNullValue = true;
```

Comment:
Shouldn't `valueIsNull` be initialized on first null to reduce builder memory footprint?

Reply:
Yes that could be done also.  Maybe some good followup work.